py/dml/structure.py

```diff
@@ -1426,12 +1439,21 @@ def process_method_implementations(obj, name, implementations,
                     raise EMETH(impl.site, overridden.site,
                                 "different nothrow annotations")
 
+        template = (impl.obj_spec.parent_template
+                    if isinstance(impl.obj_spec, InstantiatedTemplateSpec)
+                    else None)
```

Comment:
This one thing is the entire reason I've refactored `MethodHandle`s  to carry around `obj_spec`:s rather than `rank`s.

Reply:
And is also why `InstantiatedTemplateSpec` exists.